src/encoding.ts

```diff
@@ -48,12 +49,12 @@ export function encodeHash(text: string): string {
  * Encode characters that need to be encoded query values on the query
  * section of the URL.
  *
- * @param text - string to encode
+ * @param input - string to encode
  * @returns encoded string
  */
-export function encodeQueryValue(text: string | number): string {
+export function encodeQueryValue(input: QueryValue): string {
   return (
-    encode(text)
+    encode(typeof input === "string" ? input : JSON.stringify(input))
```

Comment:
Would it make sense to allow users to supply their own parse/encode function?

Reply:
Then they can already stringify the values before passing to withQuery and other utils.